src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleCreationExpression.cs

```diff
@@ -59,7 +56,18 @@ private BoundNode RewriteTupleCreationExpression(BoundTupleCreationExpression no
                                                                                               TupleTypeSymbol.RestPosition - 1)
                                                                                       .Add(currentCreation);
 
-                        MethodSymbol constructor = tuple8Ctor.AsMember(underlyingTupleTypeChain.Pop());
+                        NamedTypeSymbol currentType = underlyingTupleTypeChain.Pop();
+                        var tuple8Ctor = (MethodSymbol)TupleTypeSymbol.GetWellKnownMemberInType(currentType.OriginalDefinition,
+                                                                                                TupleTypeSymbol.GetTupleCtor(TupleTypeSymbol.RestPosition),
+                                                                                                _compilation.Assembly,
+                                                                                                _diagnostics,
+                                                                                                node.Syntax);
+                        if ((object)tuple8Ctor == null)
+                        {
+                            return node;
+                        }
```

Comment:
I think we should take a chance here and require the types be consistent throughout an underlying type, we will enforce this consistency in metadata import and already enforce it for source. Besides, you are already taking this chance with fields.